packages/client/src/features/smart-connector/smart-connector.ts

```diff
@@ -0,0 +1,524 @@
+        group.appendChild(groupItems);
+        this.groupIsTop[group.id] = false;
+        return group;
+    }
+
+    protected createGroupHeader(group: SmartConnectorGroupItem, groupItems: HTMLElement, searchField: HTMLElement): HTMLElement{
+        const header = document.createElement('div');
+        const headerTitle = document.createElement('div');
+        header.classList.add(HEADER_CLASS);
+        // for same css as palette header
+        header.classList.add('group-header');
+        if (group.icon) {
+            headerTitle.appendChild(createIcon(group.icon));
+        }
+        headerTitle.insertAdjacentText('beforeend', group.label);
+        headerTitle.classList.add('header-title');
+        header.appendChild(headerTitle);
+        header.tabIndex = 0;
+        if (group.submenu) {
+            const submenuIcon = group.position === SmartConnectorPosition.Top ? createIcon('chevron-up') : createIcon('chevron-down');
+            header.appendChild(submenuIcon);
+            groupItems.classList.add(COLLAPSABLE_CLASS);
+            header.onclick = _ev => {
+                this.toggleSubmenu(submenuIcon, group, groupItems, searchField);
+            };
+            header.onkeydown = ev => {
+                if (matchesKeystroke(ev, 'Enter')) {
+                    this.toggleSubmenu(submenuIcon, group, groupItems, searchField);
+                }
+                this.headerNavigation(ev, groupItems, header);
+            };
+            this.groupIsCollapsed[group.id] = true;
+        }
+        return header;
+    }
+
+    protected toggleSubmenu(icon: HTMLElement, group: SmartConnectorGroupItem, groupItems: HTMLElement, searchField: HTMLElement): void {
+        changeCodiconClass(icon, 'chevron-up');
+        changeCodiconClass(icon, 'chevron-down');
+        this.groupIsCollapsed[group.id] = !this.groupIsCollapsed[group.id];
+        if (groupItems.style.maxHeight) {
+            groupItems.style.maxHeight = '';
+            searchField.style.maxHeight = '';
+        } else {
+            groupItems.style.maxHeight = MAX_HEIGHT_GROUP;
+            searchField.style.maxHeight = '50px';
+        }
+    }
+
+    protected createToolButton(item: PaletteItem): HTMLElement {
+        const button = document.createElement('div');
+        button.tabIndex = 0;
+        button.classList.add(TOOL_BUTTON_CLASS);
+        if (item.icon) {
+            button.appendChild(createIcon(item.icon));
+            return button;
+        }
+        button.insertAdjacentText('beforeend', item.label);
+        button.onclick = this.onClickCreateToolButton(button, item);
+        button.onkeydown = ev => this.toolButtonKeyHandler(ev, item);
+        button.id = item.id;
+        return button;
+    }
+
+    protected createSearchField(itemGroup: SmartConnectorGroupItem): HTMLElement {
+        const searchField = document.createElement('input');
+        searchField.classList.add('smart-connector-search');
```

Comment:
Please extract that string to the `consts` defined before as you already did in other places here.